agent/bench-scripts/driver/linpack

```diff
@@ -0,0 +1,333 @@
+fi
+output_dir=${output_dir:-$(pwd)}
+if [[ ! -d "${output_dir}" ]]; then
+	printf -- "[%s] ERROR: Specified --output-dir '${output_dir}' is not a directory\n" "${script_name}" >&2
+	exit 2
+fi
+use_omp=${use_omp:-${use_omp_def}}
+if [[ "${use_omp}" != "y" && "${use_omp}" != "n" ]]; then
+	printf -- "[%s] ERROR: --use-omp must either be 'y' or 'n'\n" "${script_name}" >&2
+	exit 2
+fi
+if [[ "${use_omp}" == "n" && (-n "${threads_arg}" || -n "${kmp_affinity_args}") ]]; then
+	printf -- "[%s] ERROR: --threads and --kmp-affinity not allowed with --use-omp=n\n" "${script_name}" >&2
+	exit 2
+fi
+
+problem_sizes=${problem_sizes:-${problem_sizes_def}}
+ps_list=(${problem_sizes})
+leading_dimensions=${leading_dimensions:-${leading_dimensions_def}}
+ls_list=(${leading_dimensions})
+run_trials=${run_trials:-${run_trials_def}}
+rt_list=(${run_trials})
+alignment_values=${alignment_values:-${alignment_values_def}}
+av_list=(${alignment_values})
+number_of_tests=${#ps_list[*]}
+if [[ ${#ls_list[*]} -ne ${number_of_tests} || \
+      ${#rt_list[*]} -ne ${number_of_tests} || \
+      ${#av_list[*]} -ne ${number_of_tests} ]]; then
+	printf -- "[%s] ERROR: You must have the same number of elements for --problem-sizes, --leading-dimensions, --run-trials, and --alignment-values\n" >&2
+	exit 2
+fi
+
+linpack_input_file="${output_dir}/${linpack_input_file_name}"
+linpack_metadata_file="${output_dir}/${linpack_metadata_file_name}"
+linpack_output_file="${output_dir}/${linpack_output_file_name}"
+linpack_command_file="${output_dir}/${linpack_command_file_name}"
+
+# N.B. - Trailing whitespace is required.
+numactl_cmd=${numactl_args:+"numactl ${numactl_args} "}
+
+if [[ "${use_omp}" == "y" ]]; then
+	threads_arg=${threads_arg:-$(grep -c processor /proc/cpuinfo)}
+	kmp_affinity_args=${kmp_affinity_args:-${kmp_affinity_args_def}}
+
+	# N.B. - Trailing whitespace is required.
+	omp_num_threads="OMP_NUM_THREADS=${threads_arg} "
+	# N.B. - Trailing whitespace is required.
+	kmp_affinity="KMP_AFFINITY=${kmp_affinity_args} "
+else
+	# No environment variables to add to the command line.
+	omp_num_threads=""
+	kmp_affinity=""
+fi
+final_cmd="${omp_num_threads}${kmp_affinity}${numactl_cmd}${binary}"
+display_cmd="${omp_num_threads}${kmp_affinity}${numactl_cmd}${binary}"
+final_cmd+=" ${linpack_input_file} |"
+final_cmd+=" stdbuf --input=0 tee -i ${linpack_output_file}"
+# Save the actual command we'll use below
+printf -- "${final_cmd}\n" > ${linpack_command_file}
+
+# Line 1, providing a default header (truncated to 254 characters below)
+header=${header:-"Pbench - LINPACK Benchmark data file (${linpack_input_file})"}
+# Line 2, providing a default sub-header (truncated to 254 characters below)
+subheader=${subheader:-"Pbench - LINPACK Command: '${display_cmd}'"}
+
+# Create the LINPACK input file.  Note the comment lines are redundant but
+# provided in a spirit of consistency with the example input files provided by
+# linpack RPM.
+cat > ${linpack_input_file} <<-__EOF__
```

Comment:
Just as here, the lines *are* tab-indented so the `<<-` dedents the text... cool.